### ISSUE TITLE
Regression in undesirable_function_linter

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#866]
Reported by @jennybc at de3fb6b730c10f4ee2b222e1d58a5dcbc9604b68:

> This change broke my use of the undesirable function lintr in GHA:
> 
> https://github.com/r-lib/usethis/actions/workflows/lint-undesirable-functions.yaml
> 
> Specifically, here's what little info I get:
> 
> ```
> Error: Error in source_file$lines[[line]] : subscript out of bounds
> Calls: <Anonymous> ... assign_item -> <Anonymous> -> lapply -> FUN -> Lint -> structure
> ```
> 
> If I install from the previous commit, everything works normally.

I'm looking at this now. I found the error is triggered by:

testthat/test-git-default-branch.R

Investigating why...

**Comment by jennybc** [comment]
FYI, I've fixed that file now, i.e. it longer has a lint. But before I did that, I looked at this with `options(error = recover)` and it seems to have something to do with 

https://github.com/jimhester/lintr/blob/de3fb6b730c10f4ee2b222e1d58a5dcbc9604b68/R/undesirable_function_linter.R#L50
